contrib/lua/premake5.lua

```diff
@@ -21,8 +21,8 @@ project "lua-lib"
 		"etc/*.c"
 	}
 
-	configuration "linux or bsd or hurd or aix"
+	filter "system:linux or system:bsd or system:hurd or system:aix"
```

Comment:
Okay everyone, we're getting off topic here—this PR is only to bring `premake5.lua` up to date. Let's get it merged with `filter "system:linux or bsd or hurd or aix"`, and open a new issue or discussion for the criteria formatting.